### PR TITLE
Issue 86

### PR DESCRIPTION
**Pull request by mwerwee**
Fixed standard Urbain model and made sure Riboud model is correct. The dataset needs to be re-evaluated.
Added the Modified Urbain model.

Need to add reference for modified urbain model to bibtex file

### CODE REVIEW

File: src/tools/materialphysicalproperties/slags.py [hunk 1]
```diff
@@ -127,12 +125,129 @@ def calculate(self, **state):
...
+                         state_schema, None, None)
+
+    def calculate(self, **state):
+        """
+        Calculate dynamic viscosity at the specified temperature and
+        composition:
+
+        :param T: [K] temperature
+        :param x: [mole fraction] composition dictionary , e.g. \
+        {'SiO2': 0.25, 'CaO': 0.25, 'MgO': 0.25, 'FeO': 0.25}
```
Comment (christoffkok) [inline src/tools/materialphysicalproperties/slags.py]: There is also a way to present write multi-line :param:" descriptions:
:param x: [mole fraction] composition dictionary , e.g.
  {'SiO2': 0.25, 'CaO': 0.25, 'MgO': 0.25, 'FeO': 0.25}
Identing the "{'SiO2': 0.25, 'CaO': 0.25, 'MgO': 0.25, 'FeO': 0.25}" line by 2 spaces will ensure that it is displayed nicely as a 2nd line in 'help()' and sphinx api documentation.

File: src/examples/tools/materialphysicalproperties/slags.ipynb [hunk 2]
```diff
@@ -55,11 +55,23 @@
   },
   {
    "cell_type": "code",
-   "execution_count": null,
+   "execution_count": 1,
    "metadata": {
     "collapsed": false
    },
-   "outputs": [],
+   "outputs": [
```
Comment (christoffkok) [inline src/examples/tools/materialphysicalproperties/slags.ipynb]: It looks like you saved the Jupyter notebook with output. You can clear the output and then save it.

File: src/tools/materialphysicalproperties/slags.py [hunk 3]
```diff
@@ -230,4 +347,4 @@ def calculate(self, **state):
         return super().calculate(T=T, x=x)
 
 
-ds1 = DataSet(_path('data/dataset-slag-slagatlas1995.csv'))
+ds1 = DataSet(_path('data/dataset-slag-slagatlas1995.csv'))
```
Comment (christoffkok) [inline src/tools/materialphysicalproperties/slags.py]: The pep-8 standards insists on a empty new line at the end of the file

**Comment by christoffkok** [comment]
Hi Markus,

The quality of your code is really good!

I am waiting for Johan to review your changes as my metallurgy knowledge is terrible to non-existent. He won't be able to do it today. We will add it to next week's release.

I've made some comments on the changes you made. Thank you for the good quality code.

Regards,
Christoff

File: src/tools/materialphysicalproperties/slags.py [hunk 4]
```diff
@@ -127,12 +125,129 @@ def calculate(self, **state):
...
+    slag as a function of temperature and composition expressed in mole
+    fraction. This "Modified" Urbain Model is more appropriate for Basic
+    Oxygen Steelmaking Slags than the standard, generic UrbainViscosity model.
+    """
+
+    def __init__(self):
+        state_schema = {'T': {'required': True, 'type': 'float', 'min': 0.0},
+                        'x': {'required': True, 'type': 'dict'}}
+        super().__init__('Slag', 'Dynamic Viscosity', 'mu', '\\mu', 'Pa.s',
+                         state_schema, None, None)
```
Comment (johanzietsman-em) [inline src/tools/materialphysicalproperties/slags.py]: Please pass values for the `references` and `datasets` parameters to the base class constructor for the following models in this module:

- [ ] UrbainViscosityTx
- [ ] RiboudViscosityTx
- [ ] ModifiedUrbainViscosityTx

For an example of how to do this, refer to the `auxi.tools.materialphysicalproperties.core.Model` class docstring.

**Comment by johanzietsman-em** [comment]
Hi Markus

Thanks for the contribution. It looks great. I added a note above about references and data sets. Please address this. I also responded to your question on the forum about how to add your own references and data sets.

Your issue number is actually 86, and not 87. I am not sure where this misunderstanding originated. Please rename your local branch to issue-87, and address the issues that @christoffkok and I highlighted. Then push to branch issue-87 on your fork (referred to as origin). From there you can do a pull request from your branch issue-87 to the central repository (upstream) branch issue-87. Things should run smoothly from there.

Thanks

Johan

**Comment by mwerwee** [comment]
Deleting branch / closing pull request. Will change everything properly on the actual issue (Issue #86)

**Pull request closed without merging**